src/compiler/checker.ts

```diff
@@ -5754,6 +5758,22 @@ namespace ts {
             return getNarrowedTypeOfSymbol(getExportSymbolOfValueSymbolIfExported(symbol), node);
         }
 
+        function checkEnumIdentifier(node: Node, symbol: Symbol) {
+            // const enums are only permitted in:
+            // - 'left' in property access
+            // - 'object' in indexed access
+            // - target in rhs of import statement
+            // - in a type query
+            if (isConstEnumSymbol(symbol) &&
+                !((node.parent.kind === SyntaxKind.PropertyAccessExpression && (<PropertyAccessExpression>node.parent).expression === node) ||
+                    (node.parent.kind === SyntaxKind.ElementAccessExpression && (<ElementAccessExpression>node.parent).expression === node) ||
+                    ((node.kind === SyntaxKind.Identifier || node.kind === SyntaxKind.QualifiedName) && isInRightSideOfImportOrExportAssignment(<Identifier>node)) ||
+                    isInTypeQuery(node))
+                ) {

```

Comment:
Format this condition similar to how it was before. Namely:

1. Two separate `if` statements, one for isConstEnumSymbol, and one for the rest of the condition.
2. Either store the latter condition in a local, or put the paren and open brace on the same line as the end of the condition. If the condition is multiple lines, put a newline before the first statement in the 'then' block.